### PR TITLE
feat(drive): convert Markdown uploads and strip YAML frontmatter

### PR DESCRIPTION
**Pull request by johnbenjaminlewis**
## **Summary**

* **Markdown to Doc Conversion:** Treat `.md` as convertible to native Google Docs when using `gog drive upload --convert`. (Uses `application/vnd.google-apps.document` metadata).  
* **Frontmatter Handling:** Strips leading YAML frontmatter (`---` ... `---`) by default to ensure clean Drive imports.  
* **New Flag:** Added `--keep-frontmatter` for users who want to preserve the header.  
* **Implementation:** Frontmatter removal is delimiter-based (no heavy YAML dependency). Centralized `---` literal in `internal/cmd/literals.go` to satisfy `goconst`.  
* **Documentation:** Updated README and `docs/spec.md` to reflect new flags and existing Google Doc → Markdown export logic.

## **User-facing changes**

* `gog drive upload file.md --convert`: Creates a Google Doc; strips frontmatter.  
* `gog drive upload file.md --convert --keep-frontmatter`: Creates a Google Doc; keeps frontmatter.

## **Test plan**

* \[x\] `make ci`  
* \[x\] **Unit tests:** Validated `stripYAMLFrontmatter`, MIME conversion logic, and edge cases.  
* \[x\] **Manual:** Verified against live Drive API (Doc creation and frontmatter stripping confirmed).

Relates to \#272 (Provides a workaround via `drive upload` import).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -892,9 +894,11 @@ gog drive upload ./path/to/file --replace <fileId>  # Replace file content in-pl
 gog drive upload ./report.docx --convert
 gog drive upload ./chart.png --convert-to sheet
 gog drive upload ./report.docx --convert --name report.docx
+gog drive upload ./notes.md --convert                              # Markdown → Google Doc (or use --convert-to doc)
 gog drive download <fileId> --out ./downloaded.bin
 gog drive download <fileId> --format pdf --out ./exported.pdf     # Google Workspace files only
 gog drive download <fileId> --format docx --out ./doc.docx
+gog drive download <fileId> --format md --out ./note.md            # Google Doc → Markdown
```
Comment (johnbenjaminlewis) [inline README.md:923]: Note: The addition of gog drive download --format md ... to the README is for documentation completeness. I’ve included it here so that the Markdown import/export flows are documented together in the same section.

(The download as md feature was originally added in #282)

**Comment by steipete** [comment]
Landed in 5d332f7 after a polish pass.

Changes added while landing:
- byte-based frontmatter stripping with BOM/CRLF coverage
- upload-path tests that inspect the actual multipart media body
- `--keep-frontmatter` live-smoke verification
- CHANGELOG entry with contributor credit

Verified locally with focused upload/frontmatter tests, `make lint`, `make test`, `make build`, and `make ci`. Also live-tested against clawdbot@gmail.com by converting Markdown to a Google Doc, exporting Markdown to verify stripping, testing the opt-out path, and trashing both test files.

Thanks again @johnbenjaminlewis.
